app/routes/account/history.tsx

```diff
@@ -9,55 +9,97 @@ import { Select } from "~/components/Select";
 import { useState } from "react";
 import { Input } from "~/components/Input";
 import { Button } from "~/components/Button";
+import { OrderListOptions, SortOrder } from "~/generated/graphql";
+
+
+
+const orderPaginationLimitSchema = z
+    .number({
+        required_error: "Limit is required",
+        invalid_type_error: "Limit must be a number",
+        coerce: true,
+    })
+    .min(10, { message: 'Limit must be at least 10' })
+    .max(50, { message: 'Maximum limit is 50' })
+    .default(10);
+
+const orderPaginationPageSchema = z
+    .number({
+        required_error: "Page is required",
+        invalid_type_error: "Page must be a number",
+        coerce: true,
+    })
+    .min(1, { message: 'Page must be at least 1' })
+    .max(1000, { message: "Page can't be over 1000" })
+    .default(1); // One based index
+
+const orderPaginationSchema = z
+    .object({
+        limit: orderPaginationLimitSchema,
+        page: orderPaginationPageSchema,
+    })
+    .default({});
+
+export const orderPaginationValidator = withZod(orderPaginationSchema);
 
 
 
 export async function loader({ request }: DataFunctionArgs) {
-    const res = await getActiveCustomerOrderList({ request });
+    const url = new URL(request.url);
+    // Careful params are user controllable data - never blindly trust it!
+    // Use the .default fallbacks in case that params are undefined i.e. `null`
+    const limit = url.searchParams.get("limit") ?? orderPaginationLimitSchema.parse(undefined);
+    const page = url.searchParams.get("page") ?? orderPaginationPageSchema.parse(undefined); 
+
+    // Validate, if we fail we redirect to default params
+    // We could provide error information but under normal usage this shouldnt happen because
+    // we also validate on client side, which means we should only land here if the user
+    // opens a manually modified or no longer supported url
+    const zodResult = orderPaginationSchema.safeParse({ limit, page });
+    if (!zodResult.success) {
+        url.searchParams.delete("limit");
+        url.searchParams.delete("page");
+        return redirect(url.toString())
+    }
```

Comment:
This is very strange, I can't see a reason why this should happen. I wasn't able to reproduce this using the wrangler environment so it might be related to Netlify. You could try just setting url.search to an empty string rather than using the delete. Another option would be to not use the redirect function and instead return a full Response object, but all of these are workarounds for an issue I cannot explain.